tests/js/spec/views/settings/project/server-side-sampling/index.spec.tsx

```diff
@@ -186,25 +141,142 @@ describe('Server-side Sampling', function () {
     });
 
     render(
-      <RouteContext.Provider
-        value={{
-          router,
-          location: router.location,
-          params: {
-            orgId: organization.slug,
-            projectId: project.slug,
-          },
-          routes: [],
-        }}
-      >
-        <OrganizationContext.Provider value={organization}>
-          <ServerSideSampling project={project} />
-        </OrganizationContext.Provider>
-      </RouteContext.Provider>
+      <TestComponent router={router} organization={organization} project={project} />
     );
 
     const deleteButtons = screen.getAllByLabelText('Delete');
     expect(deleteButtons[0]).not.toHaveAttribute('disabled'); // eslint-disable-line jest-dom/prefer-enabled-disabled
     expect(deleteButtons[1]).toHaveAttribute('disabled'); // eslint-disable-line jest-dom/prefer-enabled-disabled
   });
+
+  it('display "update sdk versions" alert and open "recommended next step" modal', async function () {
+    jest.spyOn(modal, 'openModal');
+
+    const {organization, projects, router} = getMockData({
+      projects: mockedProjects,
+    });
+
+    render(
+      <TestComponent organization={organization} project={projects[2]} router={router} />
+    );
+
+    const recommendedSdkUpgradesAlert = await screen.findByTestId(
+      'recommended-sdk-upgrades-alert'
+    );
+
+    expect(
+      within(recommendedSdkUpgradesAlert).getByText(
+        'To keep a consistent amount of transactions across your applications multiple services, we recommend you update the SDK versions for the following projects:'
+      )
+    ).toBeInTheDocument();
+
+    expect(
+      within(recommendedSdkUpgradesAlert).getByRole('link', {
+        name: mockedProjects[1].slug,
+      })
+    ).toHaveAttribute(
+      'href',
+      `/organizations/org-slug/projects/sentry/?project=${mockedProjects[1].id}`
+    );
+
+    // Open Modal
+    userEvent.click(
+      within(recommendedSdkUpgradesAlert).getByRole('button', {
+        name: 'Learn More',
+      })
+    );
+
+    expect(openModal).toHaveBeenCalled();
```

Comment:
Should we assert that the title of the modal is on screen instead?

Reply:
then I have to render the modal UI and thought that we want to have this separate